source/MaterialXGenOsl/OslNetworkShaderGenerator.cpp

```diff
@@ -0,0 +1,211 @@
+//
+// Copyright Contributors to the MaterialX Project
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <MaterialXGenOsl/OslNetworkShaderGenerator.h>
+#include <MaterialXGenOsl/OslNetworkSyntax.h>
+#include <MaterialXGenOsl/Nodes/OsoNode.h>
+
+#include <MaterialXGenShader/GenContext.h>
+#include <MaterialXGenShader/Shader.h>
+#include <MaterialXGenShader/TypeDesc.h>
+#include <MaterialXGenShader/ShaderStage.h>
+
+
+MATERIALX_NAMESPACE_BEGIN
+
+const string OslNetworkShaderGenerator::TARGET = "genoslnetwork";
+
+//
+// OslNetworkShaderGenerator methods
+//
+
+OslNetworkShaderGenerator::OslNetworkShaderGenerator(TypeSystemPtr typeSystem) :
+    ShaderGenerator(typeSystem, OslNetworkSyntax::create(typeSystem))
+{
+}
+
+ShaderNodeImplPtr OslNetworkShaderGenerator::createShaderNodeImplForImplementation(const NodeDef& /* nodedef */) const
+{
+    return OsoNode::create();
+}
+
+static string paramString(const string& paramType, const string& paramName, const string& paramValue)
+{
+    return "param " + paramType + " " + paramName + " " + paramValue + " ;";
+}
+
+static string connectString(const string& fromNode, const string& fromName, const string& toNode, const string& toName)
+{
+    return "connect " + fromNode + "." + fromName + " " + toNode + "." + toName + " ;";
+}
+
+ShaderPtr OslNetworkShaderGenerator::generate(const string& name, ElementPtr element, GenContext& context) const
+{
+    ShaderPtr shader = createShader(name, element, context);
+    ShaderGraph& graph = shader->getGraph();
+    ShaderStage& stage = shader->getStage(Stage::PIXEL);
+
+    ConstDocumentPtr document = element->getDocument();
+
+    string lastNodeName;
+    ShaderOutput* lastOutput = nullptr;
+    std::vector<string> connections;
+
+    std::set<std::string> osoPaths;
+
+    // Walk the node graph, emitting shaders and param declarations.
+    for (auto&& node : graph.getNodes()) {
```

Comment:
Just a minor note, but make sure to run `clang-format` on this new source file, so that it's aligned with the Allman braces style of the broader codebase.